src/commands/version.ts

```diff
@@ -1,39 +1,59 @@
-import {Command, Flags} from '@oclif/core'
+import {CliUx, Command, Flags} from '@oclif/core'
 import {EOL, type as osType, release as osRelease} from 'node:os'
+import {exec} from 'shelljs'
+
+interface VersionDetail {
+  cliVersion: string;
+  architecture: string;
+  nodeVersion: string;
+  pluginVersions?: string[];
+  osVersion?: string;
+}
 
 export default class Version extends Command {
+  static enableJsonFlag = true
+
   public static flags = {
     verbose: Flags.boolean({
       summary: 'Show additional information about the CLI.',
       description: 'Additionally shows the architecture, node version, operating system, and versions of plugins that the CLI is using.',
     }),
   }
 
-  async run(): Promise<void> {
+  async run(): Promise<VersionDetail> {
     const {flags} = await this.parse(Version)
-    let output = `${this.config.userAgent}${EOL}`
 
-    if (flags.verbose) {
-      const versions = this.config.userAgent.split(' ')
-      const cliVersion: string = versions[0]
-      const architecture: string = versions[1]
-      const nodeVersion: string = versions[2]
+    const versions = this.config.userAgent.split(' ')
+    const cliVersion = versions[0]
+    const architecture = versions[1]
+    const nodeVersion = versions[2]
 
-      let pluginVersions = ''
-      for (const plugin of this.config.plugins) {
-        pluginVersions += `${EOL}\t${plugin.name}@${plugin.version} (${plugin.type})`
-      }
+    const versionDetail:VersionDetail = {cliVersion, architecture, nodeVersion}
+    let output = `${cliVersion} ${architecture} ${nodeVersion}`
 
+    if (flags.verbose) {
+      const pluginVersion: string = exec('./bin/run plugins --core', {
```

Comment:
why run `plugins --core` instead of using `this.config.plugins`? I'm concerned that executing `./bin/run` won't work in the real world. 

If you need to filter out non-core plugins, then you could filter based on the `type`, e.g.

```typescript
const plugins = this.config.plugins.filter(p => p.type === 'core')
```

Reply:
+1 on this, this won't work for CLIs that don't have `plugin-plugins` installed either.